frontend/src/metabase/query_builder/selectors.js

```diff
@@ -220,10 +220,14 @@ export const getTableMetadata = createSelector(
   (tableId, metadata) => metadata.table(tableId),
 );
 
-export const getTableForeignKeys = createSelector(
-  [getTableMetadata],
-  table => table?.fks ?? [],
-);
+export const getTableForeignKeys = createSelector([getTableMetadata], table => {
+  const tableForeignKeys = table?.fks ?? [];
+  const tableForeignKeysWithoutHiddenTables = tableForeignKeys.filter(
+    tableForeignKey => tableForeignKey.origin != null,
```

Comment:
This is the fix. For hidden tables, `ferignKey.origin` would be `undefined`. https://github.com/metabase/metabase/blob/1a3decaadadc8d2d19699857e570bbab94b362cf/frontend/src/metabase/selectors/metadata.ts#L345

We should filter out the FK foreign key completely so other parts of the code behave as if the foreign key for the hidden table never existed.

The [other suggested fix](https://github.com/metabase/metabase/issues/32654#issuecomment-1662929055) doesn't completely fix the issue since there are other places that read this `tableForeignKeys` and assume all foreign keys will have `origin` value.